src/test/java/picard/sam/markduplicates/UmiAwareMarkDuplicatesWithMateCigarTest.java

```
@@ -0,0 +1,155 @@
+                        Arrays.asList(new Boolean[] {false, true, true, true, true, true, true, true}), // All mate pairs should be duplicates except the first
+                        1
+                },
+                {
+                        // Test short UMIs with no allowance for errors
+                        Arrays.asList(new String[] {"A", "A", "T", "G", "G", "C", "C", "A"}),
+                        Arrays.asList(new String[] {"A", "A", "T", "G", "G", "C", "C", "A"}), // No UMIs should get corrected
+                        Arrays.asList(new Boolean[] {false, true, false, false, true, false, true, true}), // Only exactly duplicated UMIs will give rise to a new duplicate set
+                        0
+                },
+                {
+                        // Test longish UMIs with relatively large allowance for error
+                        // UMIs "TTGACATCCA", "ATGCCATCGA", "AAGTCACCGT" should belong to the same duplicate set since
+                        // they are within edit distance of 4 of each other.  TTGACATCCA should be chosen as the inferred
+                        // UMI even though it only occurs once.  Since all UMIs only occur once, we choose the UMI that
+                        // is not marked as duplicate to be the inferred UMI.
+                        Arrays.asList(new String[] {"TTGACATCCA", "ATGCCATCGA", "AAGTCACCGT"}),
+                        Arrays.asList(new String[] {"TTGACATCCA", "TTGACATCCA", "TTGACATCCA"}), // All UMIs should get corrected to TTGACATCCA
+                        Arrays.asList(new Boolean[] {false, true, true}), // All mate pairs should be duplicates except the first
+                        4
+                },
+                {
+                        // Test to make sure that if any reads don't have a UMI, we treat things as if there were no
+                        // UMIs at all
```

I think that if your reads don't have UMIs, you shouldn't be using this tool, and the tool should throw an error.